packages/lib/src/__tests__/basic-io.test.ts

```diff
@@ -0,0 +1,189 @@
+import { CLIError } from '@oclif/errors'
+
+import { inputAndOutputItem, Naming, outputItem, outputList, Sorting } from '../basic-io'
+import * as format from '../format'
+import { InputProcessor, UserInputProcessor } from '../input'
+import * as inputBuilder from '../input-builder'
+import { IOFormat } from '../io-util'
+import * as output from '../output'
+import * as outputBuilder from '../output-builder'
+import { SmartThingsCommandInterface } from '../smartthings-command'
+import { buildMockCommand } from './test-lib/mock-command'
+import { SimpleType } from './test-lib/simple-type'
+
+
+describe('basic-io', () => {
+	const item = { 'str': 'string', num: 5 }
+	const list = [item]
+	let baseCommand: SmartThingsCommandInterface
+	let formatAndWriteItemSpy: jest.SpyInstance<Promise<void>, [command: (SmartThingsCommandInterface & format.TableCommonOutputProducer<unknown>) | (SmartThingsCommandInterface & format.CustomCommonOutputProducer<unknown>), item: unknown, defaultIOFormat?: IOFormat | undefined]>
+	let formatAndWriteListSpy: jest.SpyInstance<Promise<void>, [command: (SmartThingsCommandInterface & Sorting & Naming) | (SmartThingsCommandInterface & format.TableCommonListOutputProducer<unknown> & Naming) | (SmartThingsCommandInterface & format.CustomCommonListOutputProducer<unknown> & Naming), list: unknown[], includeIndex?: boolean | undefined]>
```

Comment:
These lines are quite verbose. I thought the type safety would be covered by the original function declaration. Or is there some jest functionality that can only be used with these declarations in place?

Reply:
With `let` and no immediate assignment, the compiler has no way of determining the type. I know they're ugly, but VSCode figures them out for you (for spies at least).

Reply:
Can we just immediately assign then? I only see assertions on calls below which should be successfully cleared between tests with `jest.clearAllMocks()`

Reply:
Apparently I can...I had it in my head that `clearAllMocks` meant that I had to start *completely* over, even with the calling `jest.fn` or `jest.spyOn` but testing proves that not to be the case--I've updated these to assign immediately and use const.